extension-api/app/core/subscription.py

```
@@ -47,45 +49,49 @@ async def get_user_subscription(user_id: str) -> Optional[Dict[str, Any]]:
+                if plan and plan.get("limits"):
                     # Ensure limits is properly parsed as JSON
-                    limits = plan['limits']
+                    limits = plan["limits"]
                     if isinstance(limits, str):
                         import json
+
                         try:
                             limits = json.loads(limits)
                             logger.info(f"🔍 Parsed limits from JSON string: {limits}")
                         except json.JSONDecodeError as e:
                             logger.error(f"❌ Failed to parse limits JSON: {e}")
```

Ruff (0.13.3)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)